### DISCUSSION TITLE
classes/class_window

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#99] [User-contributed notes]
# classes/class_window

Inherits: Viewport< Node< Object Inherited By: AcceptDialog, Popup Base class for all windows, dialogs, and popups. Description: A node that creates a window. The window can either be a native syst...

https://docs.godotengine.org/en/stable/classes/class_window.html

**Comment by Festinocasa** [comment]
In my godot v4.3.stable.mono.official [77dcf97d8], the value of NotificationVisibilityChanged is 31, not like 30 in docs. Is that a mistake? Insteadly, 30 is a notification value can be passed in "override void _Notification(int what)" displaying no const name in my intellisense,  which is reacted to node moving.

**Reply by Festinocasa** [reply]
The docs is right. In Window class NotificationVisibilityChanged is actually 30. 31 is another case in class Cavasitem represented for NotificationVisibilityChanged which 30 is represented for NotificationDraw. There are two cases in two values.

**Comment by Samvkes** [comment]
I'm not sure if this is the correct place, but we should probably clearly document that some Window functionality (like resizing, eg: "get_window.size = Vector2(x,y)") doesn't work as expected when the window is embedded. Especially since the embedding is default behaviour now, and basically looks like a normal floating window.

**Comment by josprachi** [comment]
i want to change the color of the black bars to some other color how do i do that?

**Reply by Calinou** [reply]
[There is no built-in solution for this in 4.x](https://github.com/godotengine/godot-proposals/issues/5590), unlike in Godot 3.

This can still be accomplished by using the `expand` stretch aspect, making the root viewport to draw a plain color (or a texture), then using a SubViewport that has a fixed aspect ratio.

**Comment by admin** [comment]
*Originally posted by [**GofusProject**](https://github.com/GofusProject) on 2026-05-28 12:09:42 UTC*

> A demo showing all Window classes and their use within the main window from the official Godot repository: https://github.com/godotengine/godot-demo-projects/tree/master/misc/multiple_windows

**Reply by admin** [reply]
*Originally posted by [**Calinou**](https://github.com/Calinou) on 2026-06-09 15:31:41 UTC*

> Pull request opened:
>
> - https://github.com/godotengine/godot/pull/120117
